packages/realm/src/app-services/BaseSubscriptionSet.ts

```diff
@@ -0,0 +1,245 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2023 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import { MutableSubscriptionSet, Realm, Subscription, SubscriptionSet, assert, binding } from "../internal";
```

Comment:
Was contemplating that at first but decided to start with separate files. I'm more than happy to revisit this for the big review so that we only need to refactor this 1 time if needed.